### PR TITLE
Final additions

### PR DESCRIPTION
**Pull request by DOM-C1**
This pull request does two things. Firstly it adds the updated backend to GitHub and secondly it fixes a bug where locations would only display if there was a forecast for said location.

### CODE REVIEW

File: backend/utils_db.py [hunk 1]
```diff
@@ -40,67 +41,124 @@ def get_id(table: str, column: str, value: str, conn: connection) -> int:
         return result[0][0] if result else ERROR_CODE
```
Comment (NMcKittie) [inline backend/utils_db.py:41]: are you able to remove hardcoded values and magic numbers?

File: backend/app.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    data = request.json
+    email = data['email'].lower()
+    password = data['password']
+    if not check_row_exists(conn, 'user_details', 'email', email, 'password', password):
+        return jsonify({'error': 'Invalid credentials'}), 400
+
+    _id = get_id('user_details', 'email', email, conn)
+    name = get_value_from_db('user_details', 'name', _id, 'user_id', conn)
+    conn.close()
+    return jsonify({'message': 'Login Successful', 'name': name}), 200
```
Comment (NMcKittie) [inline backend/app.py]: you could create a global constant for the network error codes like 200. It would cut down the hardcoded numbers

Comment (DOM-C1) [reply backend/app.py]: Updated

**Review by NMcKittie — APPROVED** [review]
Looks good to me

**Pull request merged**